include/npcomp/RefBackend/Runtime/UserAPI.h

```diff
@@ -134,6 +137,7 @@ class Tensor {
   }
   // Reference count management.
   template <typename T> friend class Ref;
+  friend struct RuntimeValue;
```

Comment:
This seemed like the lesser of evils when trying to get access to the refCount from the `RuntimeValue` side of things. IValue handles this with the `c10::intrusive_ptr_target*` abstraction which gives you a runtime construct to latch onto for creating the `Union`. Otherwise things get hairy when trying to have a union with a member of `Ref<T>`

Reply:
I think it makes sense to introduce a intrusive_ptr_target-like base class (just with refcount for now; no need for weak count yet). Then we can use `Ref::incref(RefTarget *)`/`Ref::decref(RefTarget *)` from `RuntimeValue`'s methods to avoid the raw `+=` and desctruction dance inside RuntimeValue.